Source/WebCore/xml/XPathPredicate.cpp

```diff
@@ -281,7 +281,12 @@ Value Union::evaluate() const
 
 bool evaluatePredicate(const Expression& expression)
 {
-    Value result(expression.evaluate());
+    EvaluationContext clonedContext(Expression::evaluationContext());
+    Value result = [&] {
+        SetForScope<EvaluationContext> contextForScope(Expression::evaluationContext(), clonedContext);
```

Comment:
Can we not just write `SetForScope` here? If not, we should eventually create the necessary deduction guide so we can do that in the future.

Reply:
`SetForScope` works, will fix.